crates/cheatnet/src/lib.rs

```diff
@@ -14,13 +14,13 @@ pub mod state;
 
 pub struct CheatnetState {
     cheatcode_state: CheatcodeState,
-    blockifier_state: CachedState<StateReaderProxy>,
+    blockifier_state: CachedState<ExtendedStateReader>,
     pub deploy_salt_base: u32,
 }
 
 impl CheatnetState {
     #[must_use]
-    pub fn new(state: StateReaderProxy) -> Self {
```

Comment:
I'm wondering whether the `new` method should stick to creating the non-forkable state perhaps creating the `ExtendedStateReader` under the hood but I'm fine with this approach